src/utils/mdast-to-vdom.js

```diff
@@ -194,7 +195,9 @@ class VDOMTransformer {
    * @returns {string} the corresponding HTML
    */
   static toHTML(mdast, handlers) {
-    return hast2html(mdast2hast(mdast, { handlers }));
+    const rawHast = mdast2hast(mdast, { handlers });
+    const sanitizedHast = sanitize(rawHast);
```

Comment:
There are a ton of well maintained, well reviewed (just by sheer amount) html sanitizers out there. Just fom 5s of research I get:

https://github.com/cure53/DOMPurify

I would rather trust one of those – if just by sheer number of users – than a 
library that is used by few and touched every couple of months…
This is a security issue after all…

I don't see an advantage of doing this in htast over dom – but I am tooting the same horn here as in the other PR. If we model this as a dom transforming pipeline step we give users the ability to opt out (maybe they want to have js in ther markdown) and we are Independent from htast/remark.

Reply:
@koraa 

> than a library that is used by few and touched every couple of months…

I'd tend to agree here. Although this was updated in the last couple months and does have some good tests for it. 

Plus it is from the same org as a lot of our other modules (hast-to-hyperscript, hast-util-to-html, mdast-util-to-hast, mdast-util-to-string, unist-util-map, unist-util-select, …) which basically face the same issues, so my assumption was these thresholds (last commit, number of maintainers, tests, etc.) were the accepted norm.

I'd be happy to start a discussion about how to properly chose dependencies for helix. Since the project is still in its infancy, it's a good time to define best practices and choose dependencies carefully for the long run.
Would you agree to handle that in a separate issue and close the security hole for now with this?

Reply:
Yeah, but those other libraries are not as security-relevant I would say…

It does have good tests, that's true. I personally do not have specific thresholds for anything; it's more about comparing the options and finding the best choice.
How about you take a look at a couple of html sanatization libs and compare them?

Reply:
I also tend to take a look at the source code of libs and tbh I am more impressed with  DOMPurify than hast-util-sanitize (the latter uses a bit of a weird style and the former seems to be much better documented and more complete)

Btw I didn't mean to imply that hast-util-sanitize was a bad or amateurish choice; if it was the only choice we had, I'd go for it. But it isn't the only choice and I think we may be able to do better.

Reply:
Agree. That is also why I hope to get some more input from @asanso. I don't want to make a pure judgment call on the repo itself for now, and want to make sure we properly cover the XSS use case with the library.

I'm also not partial with `hast-util-sanitize`. I like the kind of discussion this PR triggers.
I'll play around with other libs this week and see what I can come up with

Reply:
++1